### ISSUE TITLE
Internal Error: Unexpected extension kind: 'InputObjectTypeDefinition'

### ISSUE THREAD

**Issue by momakes2** [facebook/relay#3324]
When building using the babel plugin, it fails with the error `Module not found: Internal Error: Unexpected extension kind: 'InputObjectTypeDefinition'`. 

![CleanShot 2021-01-16 at 06 51 48](https://user-images.githubusercontent.com/71785758/104795960-8667f380-57c7-11eb-95ad-9c74a46b938f.png)

I have only one graphql version in my yarn lock file:
```
  "resolutions": {
    "graphql": "^15.4.0"
  },
  ```
  
  And here's the schema:
  ```
  ### This file was generated by Nexus Schema
### Do not make changes to this file directly

type Chat implements Node {
  createdAt: DateTime
  iconEmoji: String
  id: GlobalID!
  spaceId: GlobalID
  title: String
}

input ChatsInput {
  spaceId: GlobalID
}

"""
A date-time string at UTC, such as 2007-12-03T10:15:30Z, compliant with the `date-time` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
"""
scalar DateTime

"""
A globally unique node id consisting of type:id
"""
scalar GlobalID

interface Node {
  id: GlobalID!
}

type Query {
  chats(input: ChatsInput): Chat
  hello(name: String): String
}

type Subscription {
  greetings: [String]
}
```

**Comment by morrys** [comment]
hi @momakes2,
your problem seems related to this:
https://github.com/facebook/relay/pull/3221/files

**Comment by momakes2** [comment]
@morrys Thanks. So the only solution is to wait until it gets merged I suppose?

**Comment by sibelius** [comment]
You can use patch package as well

**Comment by morrys** [comment]
So are you using a input types on the client schema?

**Comment by momakes2** [comment]
No I don’t have a local schema (if you mean for handling local state) it’s just that schema I sent you. Btw I also have apollo queries in this project could that be related?

-
Mo, Co-Founder of There

On January 16, 2021 at 5:02 PM, Lorenzo Di Giacomo (notifications@github.com) wrote:
So are you using a input types on the client schema?

**Comment by achilsowa** [comment]
I ran into similar issue, at least the error message was exactly the same.
The error occured when I move the generate schema.graphql from the root src/ directory to another directory (src/graphql to name it)
I spent like 4 hours staging and restoring my files before I noticed that.
I put it back to the src/ directory and everything went ok again.

So I just put it there in case it help

**Comment by mak-elena** [comment]
Hi,
I ran into exacly the same issue.

@morrys, the fix you mentioned is still not merged.
Is it going to happen?

> hi @momakes2,
> your problem seems related to this:
> https://github.com/facebook/relay/pull/3221/files

**Comment by momakes2** [comment]
> Hi,
> 
> I ran into exacly the same issue.
> 
> 
> 
> @morrys, the fix you mentioned is still not merged.
> 
> Is it going to happen?
> 
> 
> 
> > hi @momakes2,
> 
> > your problem seems related to this:
> 
> > https://github.com/facebook/relay/pull/3221/files
> 
> 

I don't think it's related to that fix though

**Comment by mak-elena** [comment]
ok, @momakes2 , you are right.
 
Actually, I managed to solve the problem already.
In my case it was caused by a duplicate of schema.graphql stored in another folder.
It is a bit strange, because the path to this second instance was not referenced in any script or config.
But anyway,  the relay compiler works fine after the file was removed.  :)

**Comment by cheema-corellian** [comment]
> In my case it was caused by a duplicate of schema.graphql stored in another folder.

Yep.  Same here.  For me, I had an older file named schema3.sql in the same folder.  The unhelpful error message would have driven me insane.  I was saved by a Google search that brought me here.
